snappiershot/serializers/optional_module_utils.py

```diff
@@ -0,0 +1,39 @@
+""" Utilities for handling optional modules """
+from types import ModuleType
+from typing import Any, Optional
+
+
+def is_pandas_object(obj: Any) -> bool:
+    """Return true if the given object is a pandas object
+
+    A special effort is made to avoid importing pandas unless it's really necessary.
+    As such, this is checked by looking for "pandas" in the __module__ attribute
+    of the object's type
+
+    """
+    return "pandas" in getattr(type(obj), "__module__", "").split(".")
```

Comment:
We might want to check if the first value is `pandas`. I think this would get tripped if an object is defined within a `pandas.py` file